app/autocomplete_mention/controller.js

```diff
@@ -0,0 +1,17 @@
+'use strict';
+
+var database = require('../../lib/mongo/');
+
+module.exports.mention = function mention(req,res) {
+  database.connect(function(err,db){
+    if(err) throw err;
+    var query = req.query.pattern;
+    db.collection('user').find({
+      name : {$regex:query}
```

Comment:
Can you try [`$text`](http://docs.mongodb.org/master/reference/operator/query/text/)?